flytekit/clients/raw.py

```diff
@@ -1,15 +1,90 @@
 from __future__ import absolute_import
+
 from grpc import insecure_channel as _insecure_channel, secure_channel as _secure_channel, RpcError as _RpcError, \
     StatusCode as _GrpcStatusCode, ssl_channel_credentials as _ssl_channel_credentials
 from flyteidl.service import admin_pb2_grpc as _admin_service
 from flytekit.common.exceptions import user as _user_exceptions
+from flytekit.configuration.platform import AUTH as _AUTH
+from flytekit.configuration.creds import (
+    CLIENT_ID as _CLIENT_ID,
+    CLIENT_CREDENTIALS_SCOPE as _SCOPE,
+)
+from flytekit.clis.sdk_in_container import basic_auth as _basic_auth
+import logging as _logging
 import six as _six
+from flytekit.configuration import creds as _creds_config, platform as _platform_config
+
+from flytekit.clis.auth import credentials as _credentials_access
+
+
+def _refresh_credentials_standard(flyte_client):
+    """
+    This function is used when the configuration value for AUTH_MODE is set to 'standard'.
+    This either fetches the existing access token or initiates the flow to request a valid access token and store it.
+    :param flyte_client: RawSynchronousFlyteClient
+    :return:
+    """
+
+    _credentials_access.get_client().refresh_access_token()
+    flyte_client.set_access_token(_credentials_access.get_client().credentials.access_token)
+
+
+def _refresh_credentials_basic(flyte_client):
+    """
+    This function is used by the _handle_rpc_error decorator, depending on the AUTH_MODE config object. This handler
+    is meant for SDK use-cases of auth (like pyflyte, or when users call SDK functions that require access to Admin,
+    like when waiting for another workflow to complete from within a task). This function uses basic auth, which means
+    the credentials for basic auth must be present from wherever this code is running.
+
+    :param flyte_client: RawSynchronousFlyteClient
+    :return:
+    """
+    auth_endpoints = _credentials_access.get_authorization_endpoints()
+    token_endpoint = auth_endpoints.token_endpoint
```

Comment:
Guessing you'll have to do something here to support relative URLs.

Reply:
Not sure what you mean?

Reply:
if the token endpoint is "/token", the code doesn't work, I think